Application/Api/Billing/Bookkeeping/ApiBasketRepayment.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+namespace SPHERE\Application\Api\Billing\Bookkeeping;
+
+use SPHERE\Application\Api\ApiTrait;
+use SPHERE\Application\Api\Dispatcher;
+use SPHERE\Application\Billing\Accounting\Creditor\Creditor;
+use SPHERE\Application\Billing\Bookkeeping\Basket\Basket;
+use SPHERE\Application\Billing\Bookkeeping\Basket\Service\Entity\TblBasketType;
+use SPHERE\Application\Billing\Bookkeeping\Invoice\Invoice;
+use SPHERE\Application\Billing\Inventory\Item\Item;
+use SPHERE\Application\Billing\Inventory\Setting\Service\Entity\TblSetting;
+use SPHERE\Application\Billing\Inventory\Setting\Setting;
+use SPHERE\Application\IApiInterface;
+use SPHERE\Common\Frontend\Ajax\Emitter\ServerEmitter;
+use SPHERE\Common\Frontend\Ajax\Pipeline;
+use SPHERE\Common\Frontend\Ajax\Template\CloseModal;
+use SPHERE\Common\Frontend\Form\IFormInterface;
+use SPHERE\Common\Frontend\Form\Repository\Field\DatePicker;
+use SPHERE\Common\Frontend\Form\Repository\Field\SelectBox;
+use SPHERE\Common\Frontend\Form\Repository\Field\TextField;
+use SPHERE\Common\Frontend\Form\Structure\Form;
+use SPHERE\Common\Frontend\Form\Structure\FormColumn;
+use SPHERE\Common\Frontend\Form\Structure\FormGroup;
+use SPHERE\Common\Frontend\Form\Structure\FormRow;
+use SPHERE\Common\Frontend\Icon\Repository\Save;
+use SPHERE\Common\Frontend\Layout\Repository\Panel;
+use SPHERE\Common\Frontend\Layout\Repository\Well;
+use SPHERE\Common\Frontend\Link\Repository\Primary;
+use SPHERE\Common\Frontend\Message\Repository\Danger;
+use SPHERE\Common\Frontend\Message\Repository\Success;
+use SPHERE\Common\Frontend\Message\Repository\Warning;
+use SPHERE\Common\Frontend\Text\Repository\Danger as DangerText;
+use SPHERE\System\Extension\Extension;
+
+/**
+ * Class ApiBasketRepayment
+ * @package SPHERE\Application\Api\Billing\Accounting
+ */
+class ApiBasketRepayment extends Extension implements IApiInterface
+{
+
+    // registered method
+    use ApiTrait;
+
+    public function exportApi($Method = '')
+    {
+        $Dispatcher = new Dispatcher(__CLASS__);
+
+        // Basket
+        $Dispatcher->registerMethod('showAddBasket');
+        $Dispatcher->registerMethod('saveAddBasket');
+
+        return $Dispatcher->callMethod($Method);
+    }
+
+    /**
+     * @param string $Identifier
+     * @param string $Type
+     * @param array  $Basket
+     * @param array  $ErrorHelp
+     *
+     * @return Pipeline
+     */
+    public static function pipelineOpenAddBasketModal($Identifier = '', $Type = '', $Basket = array(), $ErrorHelp = array())
```

Comment:
Codacy found an issue: [Incorrect spacing between argument "$Identifier" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)